lib/hmmlearn/base.py

```diff
@@ -253,6 +253,64 @@ def score(self, X, lengths=None):
         """
         return self._score(X, lengths, compute_posteriors=False)[0]
 
+    def score_batches(self, X, lengths=None):
+        """
+        Compute the log probability under the model.
+        Compared to score() it returns a list of probabilities and doesn't add them up.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Feature matrix of individual samples.
+        lengths : array-like of integers, shape (n_sequences, ), optional
+            Lengths of the individual sequences in ``X``. The sum of
+            these should be ``n_samples``.
+
+        Returns
+        -------
+        logprobs : List[float]
+            Log likelihoods of sequences in ``X``.
+
+        """
+
+        _utils.check_is_fitted(self, "startprob_")
+        self._check()
+
+        X = check_array(X)
+        impl = {
+            "scaling": self._score_batches_scaling,
+            "log": self._score_batches_log,
+        }[self.implementation]
+        return impl(X=X, lengths=lengths)
+
+    def _score_batches_log(self, X, lengths=None):
+
+        logprobs = []
+        for sub_X in _utils.split_X_lengths(X, lengths):
+            try:
+                framelogprob = self._compute_log_likelihood(sub_X)
+            except IndexError:
+                logprobij = -np.inf
```

Comment:
when does this occur?  why is this not covered by tests?

Reply:
It occurs when states not found in training are scored in `MultinomialHMM` for example. In that case it makes sense to have this specific sequence return -inf since it encountered a zero chance transition I think.

ie. `fit([[0])` `score([[1]])`

Reply:
thanks for the example, see comment above.